javalib/src/main/scala/java/lang/Long.scala

```diff
@@ -36,82 +48,339 @@ object Long {
+    if (x == y) 0
+    else if (x < y) -1
+    else 1
 
-  @inline def toString(i: scala.Long): String = ???
+  @inline def compareUnsigned(x: scala.Long, y: scala.Long): scala.Int =
+    compare(x ^ scala.Long.MinValue, y ^ scala.Long.MinValue)
+
+  def decode(nm: String): Long = {
+    val length = nm.length()
+    if (length == 0) {
+      throw new NumberFormatException()
+    } else {
+      var i          = 0
+      var firstDigit = nm.charAt(i)
+      val negative = firstDigit == '-'
+
+      if (negative) {
+        if (length == 1) {
+          throw new NumberFormatException(nm)
+        }
+
+        i += 1
+        firstDigit = nm.charAt(i)
+      }
+
+      var base = 10
+      if (firstDigit == '0') {
+        i += 1
+        if (i == length) {
+          return valueOf(0L)
+        }
+
+        firstDigit = nm.charAt(i)
+        if (firstDigit == 'x' || firstDigit == 'X') {
+          if (i == length) {
+            throw new NumberFormatException(nm)
+          }
+          i += 1
+          base = 16
+        } else {
+          base = 8
+        }
+      } else if (firstDigit == '#') {
+        if (i == length) {
+          throw new NumberFormatException(nm)
+        }
+        i += 1
+        base = 16
+      }
+
+      valueOf(parse(nm, i, base, negative))
+    }
+  }
 
-  @inline def toUnsignedString(i: scala.Long): String = ???
+  @inline
+  def divideUnsigned(dividend: scala.Long, divisor: scala.Long): scala.Long =
+    divULong(dividend, divisor)
 
-  def parseLong(s: String, radix: Int): scala.Long = ???
+  def getLong(nm: String): Long =
+    if (nm == null || nm.length() == 0) {
+      null
+    } else {
+      val prop = System.getProperty(nm)
+      if (prop == null) {
+        null
+      } else {
+        try {
+          decode(prop)
+        } catch {
+          case e: NumberFormatException =>
+            null
+        }
+      }
+    }
+
+  def getLong(nm: String, v: scala.Long): Long =
+    if (nm == null || nm.length() == 0) {
+      valueOf(v)
+    } else {
+      val prop = System.getProperty(nm)
+      if (prop == null) {
+        valueOf(v)
+      } else {
+        try {
+          decode(prop)
+        } catch {
+          case e: NumberFormatException =>
+            valueOf(v)
+        }
+      }
+    }
+
+  @inline def getLong(nm: String, v: Long): Long =
+    getLong(nm, v.longValue)
 
-  @inline def parseLong(s: String): scala.Long =
-    parseLong(s, 10)
+  @inline def hashCode(value: scala.Long): Int =
+    value.toInt ^ (value >>> 32).toInt
 
-  def parseUnsignedLong(s: String, radix: Int): scala.Long = ???
+  @inline def highestOneBit(i: scala.Long): scala.Long = {
+    val x0  = i
+    val x1  = x0 | (x0 >> 1)
+    val x2  = x1 | (x1 >> 2)
+    val x4  = x2 | (x2 >> 4)
+    val x8  = x4 | (x4 >> 8)
+    val x16 = x8 | (x8 >> 16)
+    val x32 = x16 | (x16 >> 32)
 
-  @inline def parseUnsignedLong(s: String): scala.Long =
-    parseUnsignedLong(s, 10)
+    x32 & ~(x32 >>> 1)
```

Comment:
Fixed in https://github.com/scala-native/scala-native/pull/238/commits/78016c7a52b2c5da8fa61ecb04b368437103f5ee